pkg/shortCode.go

```diff
@@ -1,25 +1,47 @@
 package shortcodes
 
 import (
+	"math/big"
 	"strings"
 
 	"github.com/zainonrails/shortcodes/internal"
 )
 
-// Takes an uint id as an argument which would be taken from a database record
+// Takes a big.Int as an argument which would be taken from a database record
 // for which you need to create a shortcode
-// example: shortcode.GenerateShortCode(1220) => "QqWZcuh"
-func GenerateShortCode(id uint) string {
+// example: shortcode.GenerateShortCode(big.NewInt(1220)) => "QqWZcuh"
+
+func GenerateShortCode(id *big.Int) string {
```

Comment:
Would it be possible to do some kind of conversion here? This format introduces dependencies on the consumers end as well, in that they now must understand how to use big ints in go just to use this library. Perhaps we could still just take a uint64 and convert that to big int?

Reply:
That was initial implementation to convert the number to big.Int but I am concerned that there could be some id/number that might overflow uint (highly unlikely) that an app would have that many records.

I think the param can be changed from uint to string and then converting that string to big.Int, that way length wouldn't matter.
@Shahab96

Reply:
I have reverted it back to `uint` type.

Reply:
If it's possible to error here we can also just perform a check and return a tuple with the error

Reply:
I don't see where we can do that. If the param is of different type or overflows, go should take care of that. Where do think we can add the check? @Shahab96